syntax = "proto3";

import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";
import "google/protobuf/struct.proto";

message Metadata {
  google.protobuf.Timestamp response_datetime = 1;
  google.protobuf.Duration request_duration = 2;
}

message Quotecast {
  string json_data = 1;
  Metadata metadata = 2;

  message Request {
    // <vwd_id, metrics>
    map<string, google.protobuf.ListValue> subscriptions = 1;
    // <vwd_id, metrics>
    map<string, google.protobuf.ListValue> unsubscriptions = 2;
  }
}

message Ticker {
  message Metrics {
    // <metric_name, metric_value>
    map<string, double> metrics = 1; 
  }
  Metadata metadata = 1;
  // <vwd_id, metrics>
  map<string, Metrics> products = 2; 
  repeated string product_list = 3;
}